cmake/Futility_CreateUnitTest.cmake

```diff
@@ -7,13 +7,43 @@
 # can be found in LICENSE.txt in the head directory of this repository.        !
 #++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
 
+FUNCTION(Futility_SetTestLabels TESTTMP)
+    # The test shall have one argument in the case Tribits failed to add the test
+    # This results in the test name being empty, which must be guarded against
+    SET(TESTNAME "INVALID")
+    SET(extra_args ${ARGN})
+    LIST(LENGTH extra_args num_args)
+    IF(${num_args} GREATER 0)
+      LIST(GET extra_args 0 TESTCAT)
+      SET(TESTNAME ${TESTTMP})
+    ENDIF()
+
+    IF(NOT "${TESTNAME}" STREQUAL "INVALID")
+      SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS ${TESTCAT})
+      IF("${TESTCAT}" STREQUAL "BASIC")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "CONTINUOUS")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "NIGHTLY")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "HEAVY")
+      ELSEIF("${TESTCAT}" STREQUAL "CONTINUOUS")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "NIGHTLY")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "HEAVY")
+      ELSEIF("${TESTCAT}" STREQUAL "NIGHTLY")
+	SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "HEAVY")
```

Comment:
So if the test being added is of category "NIGHTLY", and you execute the command ctest -L NIGHTLY, is the category the default label?

Reply:
If you execute the command ctest -L NIGHTLY then ctest will run all tests that are marked with the NIGHTLY label.

However, if you are running the NIGHTLYs it is expected you will also want the results for the BASIC and CONTINUOUS tests, so all tests added of a given category (say, BASIC) are given not only the BASIC label but also all higher labels (in this case, CONTINUOUS, NIGHTLY, HEAVY).

Hope this clears things up.

Reply:
So my question then is, do we also need:
`SET_PROPERTY(TEST ${TESTNAME} APPEND PROPERTY LABELS "NIGHTLY")`
if the category is `"NIGHTLY"`?

Reply:
We do that at line 22, the cmake code will set the "lowest" category first and then add all the higher ones explicitly in the if statements.

Reply:
Got it, sorry I missed that!